build_tools/cmake/build_benchmarks.sh

```diff
@@ -33,8 +33,8 @@ echo "Configuring to build benchmarks"
 "${CMAKE_BIN}" -B "${BUILD_BENCHMARKS_DIR}" \
   -G Ninja \
   -DIREE_HOST_BINARY_ROOT="${IREE_HOST_BINARY_ROOT}" \
-  -DIREE_BUILD_BENCHMARKS=ON \
   -DIREE_BUILD_MICROBENCHMARKS=ON \
+  -DIREE_BUILD_EXPERIMENTAL_PYTHON_GENERATED_BENCHMARKS=ON \
```

Comment:
This is a big step! Let's make sure we're ready before flipping this. Or perhaps this is just for testing during presubmit?

Reply:
`build_tools/cmake/build_benchmarks.sh` currently is only used in CI pipeline `build_suites` and for e2e model tests. So I think it's safe to flip now with the changes of e2e model tests. The downside is the old benchmark suites and the new one might be out of sync and confusing (the buildkite pipeline and CI are using different benchmark suites).

I'm fine to flip this later there is a concern.

The buildkite benchmark pipeline is using a separate script to build benchmark modules.

Reply:
But if we don't turn this on, it'll be hard to test the new benchmark pipelines, unless we are okay to build both old and new benchmark suites in CI at the same time.

Reply:
We use the buildkite benchmark result for performance regression, so having GH presubmit and buildkite out of sync can be concerning.

Maybe we can run this path in postsubmit only, create a temporary fork `iree_experimental_benchmark_suite_module_test` with the CMake implementation change, and keep the current `tests/e2e/model/CMakeLists.txt` target until we see everything is stable and we are ready to flip both github action and buildkite?

Reply:
Creating a fork in postsubmit pipeline to build the new benchmark suite will still cause the benchmark suite built twice in postsubmit, which increases the loading of runners. I'm okay to do so if we can accept the overhead.

Although personally I don't see a very high risk to flip this right now except for the concern of being out of sync. Because in presubmit we can test if the e2e model tests and buildkite pipeline works correctly.

Reply:
I'd rather not have the skew. That means we could discover a problem only on postsubmit. Let's start with the duplication, migrate to the new system, and then turn down the old one

Reply:
Sure, I'm happy that we agreed to build the benchmark suite twice during the migration : ) 
Removed any CI and test changes from this PR. Only adds the tool to generate the CMake files.